### PR TITLE
added get comments for articleByid + error handling

### PR DESCRIPTION
**Pull request by apwb89**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -105,19 +103,55 @@ describe('GET /api/articles', () => {
...
+        test('Returns 204 no content when selecting an article with no comments', ()=> {
+            return request(app)
+            .get('/api/articles/2/comments')
+            .expect(204)
+            .then((response) => {
+                expect(response.body).toEqual({});
+            })
+        })
+    })
+
```
Comment (jadelandeg) [inline __tests__/app.test.js:165]: Also need to test for an invalid article id (400)

Comment (jadelandeg) [reply __tests__/app.test.js:165]: consider sending a 200 instead of 204 for empty comments as it can cause problems client side

**Review by jadelandeg — CHANGES_REQUESTED** [review]
a couple of things to consider in testing

File: __tests__/app.test.js (hunk 1 above)
Comment (apwb89) [reply __tests__/app.test.js:165]: How  do we then indicate that there is no content? or is sending an empty array enough

**Pull request merged**